### DISCUSSION TITLE
Different ports

### DISCUSSION THREAD

**Discussion by shawwellpete** [jasonacox/Powerwall-Dashboard#152] [Q&A]
Hi Guys,

I love the look of Powerwall-Dashboard and have tried to set it up. I am running Docker and Portainer in Debian which I use for Home Assistant, TeslaMate, Grafana and a few other apps. I have created Portainer Stacks (which use Docker Compose) for all the apps.

I followed the installation instructions and got this error

ERROR: for influxdb  Cannot create container for service influxdb: Conflict. The container name "/influxdb" is already in use by container "9b9d8614419791c672a1b691a972c7d67a890192309154a0a7b6ef1b2d112e20". You have to remove (or rename) that container to be able to reuse t
Creating pypowerwall ... done

I do have an instance of influxdb which I use to allow me to use Grafana on Home Assistant data, I also already have Grafana installed for Teslamate (on port 3000). Portainer uses port 9000.

Ideally I would like to set up a  Docker Compose Stack for Powerwall-Dashboard using my current installations of Grafana and Influx-db.

I found the pypowerwalll.yml that was created by my setup process which seems to file I need. Any suggestions on how to edit it to use my current installations?

Thank you, in Advance for any help and Happy New Year!

**Comment by BJReplay** [comment]
OK, so it sounds like you already have some familiarity with Docker Compose, Grafana and InfluxDB - which is probably better than me - I'm just finding my feet, but I'll have a go...

As you've no doubt figured out, Powerwall-Dashboard uses InfluxDB to store the data, and Grafana to display it, and adds Telegraf, optionally Weather411 to pull in weather data, and of course uses PyPowerwall to get the Powerwall data.

Two options:
1. Run up a completely new stack that includes a second instance of grafana and influxdb to prove that this works
2. Modify this to use your existing grafana and influxdb (which is your preference)

I'd suggest option 1 - at least at first - and if you're happy, try option 2.

For option 1, you may only need to change `container_name: influxdbpw` under the influxdb section in powerwall.yml - you might need to do the same for grafana - if you run into a similar error for grafana having made the first change.

For option 2, deleting the entire service definition for influxdb might or might not allow this docker compose to find your existing influxdb instance and use it.

**Reply by shawwellpete** [reply]
Good ideas, I'll give them a try. One concern is the port conflict with portainer which is on 9000. Wondering if it's easier to change the port I use for portainer or change the port this uses for grafana.

**Comment by youzer-name** [comment]
@shawwellpete I'm in a similar situation. I had influxdb and telegraf running for a home automation project and another instance of Grafana running also. 

All of my powerwall containers have a 'pw-' prefix to deconflict the container names, so pw-grafana, pw-influxdb, etc. I had to use alternate ports to deconflict everything.

I'm pretty sure that within the stack, if you use the container name instead of the IP, you can re-use ports. But those ports can't be published outside the container if they conflict. So IIRC you can have two stacks, for example "powerwall-dashboard" and "home-assistant", with containers "pw-influxdb" and "influxdb", then within the powerwall-dashboard stack you can use "pw-influxdb:8086" and within the home-assistant stack you can use "influxdb:8086". 

I eventually shut down the other influx and grafana containers, but I'm still on alternate ports for both of those and portainer. At this point I don't think I have any conflicts, but it's too much bother to go and change everything back to the defaults when everything works fine the way it is.

**Reply by shawwellpete** [reply]
Thank you, that does make sense. I am now wondering if to put the work in as Home Assistant probably already gives me most of the functionality that I'd get from this. Sometimes I see a neat project and I dive in without asking if it actually helps me!
![2023-01-09_14-53-45](https://user-images.githubusercontent.com/74064242/211337110-a968c365-4b1b-4f73-bba8-a24de19f36ef.jpg)
